src/api/app/helpers/webui/webui_helper.rb

```diff
@@ -422,5 +422,9 @@ def gravatar_icon(email, size)
   def home_title
     @configuration ? @configuration['title'] : 'Open Build Service'
   end
+
+  def valid_jquery_id(id)
+    id.gsub(/([:.+])/, '\\\\\\\\\1')
```

Comment:
I don't get the part with all the backslashes (`\`). Please enlighten me :slightly_smiling_face:

Reply:
And this has nothing to do with jquery...

Reply:
> I don't get the part with all the backslashes (`\`). Why are there so many? Please enlighten me :slightly_smiling_face:

I didin't find a simpler way to escape the characters (prepend them with a backslash). Short answer: backslashes must be escaped too.

So this is what I tried for `id = tizen.org`:

```
id.gsub(/([:.+])/, '\\1')        # $('.list-group-item#tizen\1org')
id.gsub(/([:.+])/, '\\\1')       # $('.list-group-item#tizen\1org')
id.gsub(/([:.+])/, '\\\\1')      # $('.list-group-item#tizen\1org')
id.gsub(/([:.+])/, '\\\\\1')     # $('.list-group-item#tizen\.org')
id.gsub(/([:.+])/, '\\\\\\1')    # $('.list-group-item#tizen\.org')
id.gsub(/([:.+])/, '\\\\\\\1')   # $('.list-group-item#tizen\\1org')
id.gsub(/([:.+])/, '\\\\\\\\1')  # $('.list-group-item#tizen\\1org')
id.gsub(/([:.+])/, '\\\\\\\\\1') # $('.list-group-item#tizen\\.org')
```
This last one was the one that worked.

This one: `$('.list-group-item#tizen\.org')`, without escaping the backslashes, seems not to be enough.

Reply:
it might be easier to map them to _ in the output :)

Reply:
> it might be easier to map them to _ in the output :)

It might be. But this way:
- The jquerys ids aren't escaped when rendering the page. Only the id selector is escaped when rendering a UJS response.
- As `_` is also allowed in the name of projects, cases like having `tizen_org` and `tizen.org` will not cause a problem.

Reply:
I agree with @coolo. Mapping any unallowed character to `_` is simpler.

Reply:
> And this has nothing to do with jquery...

This is escaping the jquery id selectors, adding backslashes. Maybe the name of the helper can be changed. Are you referring to this, or to some other thing?

Reply:
Applied @dmarcoux suggestion of using a `data-interconnect` attribute, instead of using the `id` attribute. No need to use a helper. Thanks!